bin/check-azurerm-monitor-metric.rb

```diff
@@ -0,0 +1,305 @@
+         short: '-c N',
+         long: '--critical CRIT',
+         proc: proc { |val| val.to_i }
+
+  option :warning_under,
+         description: 'The warning threshold to check if the metric is forecasted to go under.',
+         short: '-W N',
+         long: '--warning-under WARN',
+         proc: proc { |val| val.to_i }
+
+  option :critical_under,
+         description: 'The critical threshold to check if the metric is forecasted to go under.',
+         short: '-C N',
+         long: '--critical-under CRIT',
+         proc: proc { |val| val.to_i }
+
+  def run
+    if config[:resource_id].to_s.empty? && config[:resource_name].to_s.empty?
+      unknown 'resource id or resource name/group/type/namespece and subscription id must be provided'
+    end
+
+    if !config[:critical_over] && !config[:warning_over] && !config[:critical_under] && !config[:warning_under]
+      unknown 'At least one threshold must be provided.'
+    end
+
+    if last_metric_values.empty?
+      unknown "There are no metric values for #{config[:metric]} on resource #{config[:resource_id] || config[:resource_name]} with aggregation #{config[:aggregation]}"
+    else
+      last_metric_values.each do |metric_val|
+        if config[:critical_over] && metric_val[:value] > config[:critical_over].to_f
+          critical "Metric #{metric_val[:metric_name]} is #{metric_val[:value]}"
+        elsif config[:warning_over] && metric_val[:value] > config[:warning_over].to_f
+          warning "Metric #{metric_val[:metric_name]} is #{metric_val[:value]}"
+        elsif config[:critical_under] && metric_val[:value] < config[:critical_under].to_f
+          critical "Metric #{metric_val[:metric_name]} is #{metric_val[:value]}"
+        elsif config[:warning_under] && metric_val[:value] < config[:warning_under].to_f
+          warning "Metric #{metric_val[:metric_name]} is #{metric_val[:value]}"
+        else
+          ok "Metric #{metric_val[:metric_name]} is #{metric_val[:value]}"
+        end
+      end
+    end
+  end
+
+  def last_metric_values
+    @last_metric_values ||= find_last_metric_values
+  end
+
+  def find_last_metric_values
+    values = []
+    metric_response[:value].each do |metric_resp_value|
+      name = metric_resp_value[:name] ? metric_resp_value[:name][:value] : ''
+
+      next if metric_resp_value[:timeseries].empty?
+
+      metric_resp_value[:timeseries].each do |ts|
+        ts[:data].reverse_each do |metric_value|
+          if metric_value[config[:aggregation].to_sym]
+            values << {
+              value: metric_value[config[:aggregation].to_sym].to_f,
+              metric_name: name
+            }
+
+            break
+          end
+        end
+      end
+    end
+
+    values
+  end
+
+  def metric_response
+    auth_header = if config[:use_assigned_identity]
+                    uri = URI.parse("http://localhost:#{config[:local_auth_port]}/oauth2/token?resource=https://management.azure.com/")
```

Comment:
It's been my experience with azure that they have a bunch of different portals and they change them regularly perhaps we should expose this so if they ever change it people can update their runtime options and unblock themselves without waiting for a fixed release. Not a requirement but just a thought.